src/main/kotlin/com/ampnet/reportservice/service/pojo/Transaction.kt

```diff
@@ -22,13 +38,113 @@ data class Transaction(
         transaction.amount,
         transaction.date,
         transaction.state
-    )
+    ) {
+        txDate = formatToYearMonthDayTime(date)
+        amountInEuro = getEurAmountFormatted(amount.toLong())
+        txStatus = getTransactionStatusType(type)
+    }
+
+    private fun formatToYearMonthDayTime(date: String): String {
+        val pattern = "MMM dd, yyyy HH:mm"
+        return DateTimeFormatter.ofPattern(pattern).format(ZonedDateTime.parse(date))
+    }
+
+    private fun getTransactionStatusType(type: TransactionsResponse.Transaction.Type): TransactionStatusType {
+        return when (type) {
+            TransactionsResponse.Transaction.Type.INVEST -> TransactionStatusType.PAID_OUT
+            TransactionsResponse.Transaction.Type.CANCEL_INVESTMENT -> TransactionStatusType.PAID_IN
+            TransactionsResponse.Transaction.Type.SHARE_PAYOUT -> TransactionStatusType.PAID_IN
+            TransactionsResponse.Transaction.Type.DEPOSIT -> TransactionStatusType.PAID_IN
+            TransactionsResponse.Transaction.Type.WITHDRAW -> TransactionStatusType.PAID_OUT
+            TransactionsResponse.Transaction.Type.UNRECOGNIZED -> TransactionStatusType.UNDEFINED
+        }
+    }
+
+    private fun getPercentageInProject(expectedFunding: Long?, amount: String): String? {
+        return expectedFunding?.let {
+            (TO_PERCENTAGE * amount.toLong() / expectedFunding).toString()
+        }
+    }
 }
 
-data class Transactions(
-    val transactions: List<Transaction>
+class TxSummary(
+    val transactions: List<Transaction>,
+    val userInfo: UserInfo
 ) {
-    val txAmountsSum: Long = transactions.sumByLong { it.amount.toLong() / FROM_CENTS_TO_EUROS }
+    private val transactionsByType = transactions.groupBy { it.type }
+    val period: String? = getPeriod(transactions)
+    val dateOfFinish: String? = getDateOfFinish(transactions)
+    val balance: String = getBalance(transactions)
+
+    val deposits = getEurAmountFormatted(
+        transactionsByType[TransactionsResponse.Transaction.Type.DEPOSIT]?.sumByLong {
+            it.amount.toLong()
+        }?.div(FROM_CENTS_TO_EUROS) ?: 0
```

Comment:
Extract to function: 
```kotlin
    private fun sumTransactionAmountsByType(type: TransactionsResponse.Transaction.Type): Long {
        return transactionsByType[type]?.sumByLong {
            it.amount.toLong()
        } ?: 0
    }
```

Looks to me that you are dividing from cents to euros 2 times. First time here, second time in `getEurAmountFormatted` function.